lib/network.cpp

```diff
@@ -5,51 +5,63 @@
  * returns the socket
  */
 int listen_on_port(char *ip, uint16_t port, int backlog, bool reuseport) {
-	int rc, arg_on=1;
-	struct sockaddr_in addr;
-	int sd;
-
-	// create a socket
-	if ( (sd = socket(PF_INET, SOCK_STREAM, 0)) < 0 ) {
-		proxy_error("Error on creating socket\n");
-		close(sd);
-		return -1;
-	}
-
-	// set SO_REUSEADDR
-	rc = setsockopt(sd, SOL_SOCKET,  SO_REUSEADDR, (char *)&arg_on, sizeof(arg_on));
-	if (rc < 0) {
-		proxy_error("setsockopt() failed\n");
-	}
+        int arg_on = 1;
```

Comment:
I see you removed `rc` .
That is perfectly fine, although let me explain why I often use an `rc` variable: it is easier to set break point in gdb :)
The compiler will optimize it away when compiling the non-debug version.